tests/init/neg/secondary-ctor2.scala

```diff
@@ -6,7 +6,7 @@ class A(b: B, x: Int) {
     }
     Inner().foo()
 
-    val f = () => new A(b, 3)
+    val f = () => new A(b, 3) // error: Cannot promote
```

Comment:
Why it's fine before?

Reply:
This is similar to early-promote5.scala. Previously `fields` doesn't include the parameters in the constructor of A.